packages/valory/skills/decision_maker_abci/behaviours/storage_manager.py

```diff
@@ -253,19 +270,102 @@ def _fetch_accuracy_info(self) -> Generator[None, None, bool]:
                 f"Could not retrieve data from the url {accuracy_link}. "
                 f"Received status code {response.status_code}."
             )
-            return False
+            return None
 
-        self.context.logger.info("Parsing accuracy information of the tools...")
         try:
-            self.accuracy_information = StringIO(response.body.decode())
+            tool_accuracy_data = StringIO(response.body.decode())
+            self.context.logger.info("Parsing accuracy information of the tools...")
+            return tool_accuracy_data
+
         except (ValueError, TypeError) as e:
             self.context.logger.error(
                 f"Could not parse response from ipfs server, "
                 f"the following error was encountered {type(e).__name__}: {e}"
             )
-            return False
+            return None
 
-        return True
+    def _set_accuracy_info_from_remote(self) -> Generator[None, None, None]:
+        """Set the tool accuracy info from the remote store."""
+        # Fetch accuracy information from remote
+        tool_accuracy_data = yield from self._fetch_accuracy_info()
```

Comment:
Use `wait_for_condition_with_sleep` here.

Reply:
will do

Reply:
I can't use `wait_for_condition_with_sleep` here as that expects a return type of bool, and `_fetch_accuracy_info` now returns Optional[StringIO]

Reply:
I don't think I can use wait_for_condition_with_sleep as fetch_accuracy_info no longer returns a condition

Reply:
Why did you decide to change the implementation of `_fetch_accuracy_info`?